src/components/converter.tsx

```diff
@@ -0,0 +1,149 @@
+"use client";
+
+import { useState } from "react";
+import useSWR from "swr";
+
+const URL = process.env.NEXT_PUBLIC_URL;
+
+const encodedCredentials = btoa(
+  `${process.env.NEXT_PUBLIC_USERNAME}:${process.env.NEXT_PUBLIC_PASSWORD}`
+);
+
+export default function Converter() {
+  const fetcher = async (url) => {
+    const request = await fetch(`${url}`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Basic ${encodedCredentials}`,
+      },
+    });
+
+    const data = await request.json();
+    return data;
+  };
+
+  const handleConversion = async () => {
+    if (amount && from && to) {
+      try {
+        const result = await fetcher(
+          `https://xecdapi.xe.com/v1/convert_from/?from=${from}&to=${to}&amount=${amount}`
+        );
+
+        if (result) {
+          setResult(result.to[0].mid);
+        } else {
+          console.log("Error");
+        }
+      } catch (err) {
+        console.log("Error");
+      }
+    }
+  };
+
+  const { data, isLoading, error } = useSWR(
+    `${URL}/v1/currencies?obsolete=false`,
+    fetcher
+  );
+
+  const [amount, setAmount] = useState(1);
+  const [from, setFrom] = useState("NGN");
+  const [to, setTo] = useState("USD");
+  const [result, setResult] = useState(0);
+  const [isClicked, setIsClicked] = useState(false);
+
+  console.log(isClicked);
+
+  return (
+    <div className="max-w-[70%] py-5 lg:h-[400px] px-5 rounded-xl flex flex-col mt-4 space-y-4 justify-center shadow-2xl mx-auto">
+      <div className="flex flex-col lg:grid lg:grid-cols-3 gap-3 place-content-center">
+        <div className="flex flex-col space-y-3">
+          <label htmlFor="amount">Amount</label>
+          <input
+            value={amount}
+            onChange={async (event) => {
+              if (isClicked) {
+                
+                setAmount(event.target.value);
+                await handleConversion();
+                
+              }
+
+              setAmount(event.target.value);
+            }}
+            type="number"
+            name="amount"
+            className="h-[50px] outline-none border-slate-300 rounded-md border-2 p-3"
+          />
```

Comment:
Maybe add a validation for the amount so the user doesn't enter a negative number?

Reply:
That is true.